packages/web-components/fast-components/src/design-system-provider/README.md

```diff
@@ -0,0 +1,279 @@
+```js
+import { FASTDesignSystemProvider } from "@microsoft/fast-components";
+```
+
+```html
+<fast-design-system-provider use-defaults>
+    Hello World
+</fast-design-system-provider>
+```
+
+_Design system_ properties can be overridden by setting the property or attribute on the `fast-design-system-provider`. See [FAST Design System Properties](#FAST-Design-System-Properties) for a comprehensive list of properties.
+
+**EXAMPLE: Setting design system properties**
+
+```html
+<fast-design-system-provider use-defaults background-color="#111111" design-unit="6">
+    Hello world!
+</fast-design-system-provider>
+```
+
+### Creating a Design System Provider
+
+To create a new _Design System Provider_, extend the `DesignSystemProvider` class and decorate it with the `@designSystemProvider` decorator, providing the decorator the element _tag name_ you wish to use:
+
+**EXAMPLE: Creating a custom Design System Provider**
+
+```ts
+import { DesignSystemProvider, designSystemProvider } from "@microsoft/fast-components";
+
+@designSystemProvider("fancy-design-system-provider")
+class FancyDesignSystemProvider extends DesignSystemProvider {}
+```
+
+```html
+<fancy-design-system-provider use-defaults>
+    Hello Fancy World!
+</fancy-design-system-provider>
+```
+
+#### Declaring Design System Properties
+
+Building off the above, _design system_ properties can be declared using the `@designSystemProperty` decorator.
+
+**EXAMPLE: Creating a design system property declaration**
+
+```ts
+// ..
+@designSystemProvider("fancy-design-system-provider")
+class FancyDesignSystemProvider extends DesignSystemProvider {
+    @attr("fancy-property")
+    @designSystemProperty({ cssCustomProperty: "fancy-property", default: "red" })
+    public fancyProperty;
+}
+```
+
+```html
+<fancy-design-system-provider use-defaults>
+    <p style="color: var(--fancy-property)">Red text!</p>
+</fancy-design-system-provider>
+```
+
+Declaring a _design system_ property will do several things. By default, the property name being decorated will become a CSS custom property when the value is set. This name can be customized by setting `cssCustomProperty` on the provided config object.
+
+```ts
+// Will create a CSS custom property accessible by `var(--fooBar)`
+@designSystemProperty({default: "foo"})
+public fooBar;
+```
+
+```ts
+// Will create a CSS custom property accessible by `var(--foo-bar)`
+@designSystemProperty({default: "foo", cssCustomProperty: "foo-bar"})
+public fooBar;
+```
+
+Additionally, creation of CSS custom properties can also be disabled by setting `cssCustomProperty` to `false`;
+
+```ts
+// No CSS custom property will be created
+@designSystemProperty({default: "foo", cssCustomProperty: false})
+public fooBar;
+```
+
+The default value _must_ be set through the `default` property on the supplied config. This is the value that will be applied when the `use-defaults` attribute is used and no override property is provided.
+
+### FAST Design System Properties
```

Comment:
I wonder if we should turn this into a table so we can see the css property next to the provider property. And if there's not a correspondence, maybe specify that it comes from a recipe?

Reply:
My plan was to just link to the API docs when we got them but I can convert to a table for now.